flink-libraries/flink-table/src/main/scala/org/apache/flink/table/catalog/ExternalCatalogTable.scala

```diff
@@ -37,25 +36,10 @@ import org.apache.flink.table.plan.stats.TableStats
   * @param lastAccessTime       Timestamp of last access of the table
   */
 case class ExternalCatalogTable(
-    identifier: TableIdentifier,
```

Comment:
We have tested this with our own catalog integration, and keep the table name should work. The sub catalog which representing the database layer can concat the db and table name when creating Table instance.